In [2]:
import pandas as pd

### Download the data and load it to Pandas. 

You can find the data files in the README.

In [3]:
titles = pd.read_csv('data/titles.csv')
titles.head()

title  year
0           The Rising Son  1990
1  The Thousand Plane Raid  1969
2         Crucea de piatra  1993
3                  Country  2000
4               Gaiking II  2011

In [4]:
cast = pd.read_csv('data/cast.csv', sep=",")
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4   NaN
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [5]:
release_dates = pd.read_csv('data/release_dates.csv', index_col=None,
                                      parse_dates=['date'])
release_dates.head()

title  year      country       date
0   #73, Shaanthi Nivaasa  2007        India 2007-06-15
1                 #Beings  2015      Romania 2015-01-29
2               #Declimax  2018  Netherlands 2018-01-21
3  #Ewankosau saranghaeyo  2015  Philippines 2015-01-21
4                 #Horror  2015          USA 2015-11-20

### 1. How many movies are listed in the titles dataframe?

In [6]:
num_titles= len(titles)
print(num_titles)

232330


232,330 movies

### 2. What are the earliest two films listed in the titles dataframe?

In [9]:
earliest_films = titles.sort_values(by='year').head(2)

Miss Jerry (1894) and The Startled Lover (1898)

### 3. What are the ten most common movie names of all time?

In [11]:
common_names = titles.value_counts('title').head(10)

In [14]:
print("Most common names are "+str(common_names))

Most common names are title
Hamlet                  18
Carmen                  16
Macbeth                 15
The Outsider            12
Maya                    12
Kismet                  11
The Three Musketeers    11
Freedom                 11
Temptation              11
Othello                 11
Name: count, dtype: int64


### 4. What are the 10 most common character names in movie history?

In [16]:
common_char_names = cast.value_counts('character').head(10)
print(common_char_names)

character
Himself        19600
Dancer         12096
Extra          10953
Reporter        8154
Doctor          7436
Student         7177
Policeman       6883
Party Guest     6742
Nurse           6676
Bartender       6613
Name: count, dtype: int64


### 5. Count the number of films that have been released in each decade over the history of cinema.

> Hint: You can make a new column to store what decade a movie is.

> Hint2: If you're stuck on how to do this, expand the cell below.

```python
titles['decade'] = (titles['year'] // 10) * 10
```

In [17]:
titles['decade'] = (titles['year'] // 10) * 10

In [20]:
decades = titles['decade'].value_counts().sort_index()
print(decades)

decade
1890        3
1900       37
1910     6512
1920     8797
1930    10097
1940     8576
1950    12711
1960    17515
1970    18714
1980    20243
1990    22225
2000    38713
2010    67977
2020      210
Name: count, dtype: int64


### 6. Retrieve the title and year of the first 10 rows in the titles dataframe.

In [21]:
titles.loc[:9, ['title', 'year']]

title  year
0                  The Rising Son  1990
1         The Thousand Plane Raid  1969
2                Crucea de piatra  1993
3                         Country  2000
4                      Gaiking II  2011
5                     Medusa (IV)  2015
6  The Fresh Air Will Do You Good  2008
7              Alex in Wonderland  1970
8                      L'outsider  2016
9          Do Outro Lado do Mundo  2008

### 7. Get the character names and types for the 5th to the 15th rows (inclusive) in the cast dataframe.

In [25]:
cast.loc[4:13,['character', 'type']]

character   type
4                   1988 Opera House Patron  actor
5                               Club Patron  actor
6                                   Dopeman  actor
7                                    Thug 1  actor
8                             Jaime (young)  actor
9    Actor Playing Eteocles from 'Antigone'  actor
10                   Two Palms - Ua'i Paame  actor
11                                    Chico  actor
12                                   Rapper  actor
13  Defensive Tackle - Washington Sentinels  actor

### 8. Extract the title and release date of the last 5 entries in the release_dates dataframe.

In [31]:
release_dates.tail(5)[['title','date']]

title       date
452651  xXx: State of the Union 2005-04-29
452652  xXx: State of the Union 2005-06-03
452653  xXx: State of the Union 2005-06-03
452654  xXx: State of the Union 2005-06-17
452655  xXx: State of the Union 2005-06-24

### 9. Select the year and country of the 20th row in the release_dates dataframe.

In [34]:
release_dates.iloc[20][['year', 'country']]

year          1971
country    Denmark
Name: 20, dtype: object

### 10. Print the name, character, and type for rows 100 to 105 (inclusive) in the cast dataframe.

In [36]:
cast.loc[100:105,['name', 'character','type']]

name         character   type
100       Mac 10           Himself  actor
101        10Gzy             Crazy  actor
102  Angel 11:11             Angel  actor
103  Angel 11:11  Funky Loft Guest  actor
104  12th Planet           Himself  actor
105      Nick 13     Archaeologist  actor